PopulateTools/gobierto
# Gobierto People module (first iteration)

Connects to #190.

### Review notes

I'm not deploying this code to Staging since we're taking advantage of the PostgreSQL JSONB data type to store and query unstructured data, so we would need to upgrade our PostgreSQL instances to 9.4+ to release these changes.

Please note that we were using PostgreSQL 9.6.1 in development env (the latest release available at that time) until we decide the version to go in production. Let's take it into account to keep exactly the same version in every environment. i.e. production, staging, development and test.

---

### What does this PR do?

This PR introduces the new `gobierto-people` module and implements a first iteration in the Admin (`gobierto-admin`) namespace, mainly focused in back-end and architecture. It also implements a new component called `DynamicContent`, that The DynamicContent which allows managing schema free content blocks via a collection of helpers that significantly ease the integration process.

This component is key and will be extensively used in this new module, for now. Many other details on the implementation can be found in the doc that comes with this PR: https://github.com/PopulateTools/gobierto/blob/3556c523e1e65d50f5fd9e15762d1ed5a262afcb/docs/dynamic-content-component.md

Any new strings are being localised into the three languages available now. I tried to do my best with the Catalan translations, but once again I'd need some help from my beloved @ferblape. Most of the strings are already localised, but there're some in which I had doubts 😛 

---

### How should this be manually tested?

First, run migrations and re-seed the database to make sure the new `GobiertoPeople` module is reachable via the `madrid.gobierto.dev` Site in the left sidebar.

Having http://gobierto.dev/admin/people/people as the main entry point for this feature, let's check that:

- [x] A Person can be created.
- [x] A Person can be updated.
- [x] New (dynamic content) fields can be added in this context.
- [x] A new Content block can be created.
- [x] A Content block can be updated, by changing its title and any of its fields' properties.
- [x] New fields can be added or deleted in a Content block context.
- [x] A Content block can be deleted as well.
- [x] Everything is localised.
- [x] There's a good test coverage for every new part!

---

#### ✏️ Missing stuff

- Integrate the latest front-end from https://github.com/PopulateTools/gobierto/pull/188.
- Extend the `GobiertoPeople::Person` resource by adding new fields present in the latest specification.
- Limit the Content block management options to privileged Admin users (Manager).
- Continue developing the rest of resources at `Person` level such as Events, Statements and Posts.
